aten/src/ATen/xpu/CachingHostAllocator.h

```diff
@@ -7,14 +7,20 @@
 
 namespace at::xpu {
 
-TORCH_XPU_API c10::Allocator* getCachingHostAllocator();
+inline TORCH_XPU_API at::HostAllocator* getCachingHostAllocator() {
+  return at::getHostAllocator(at::kXPU);
+}
 
-TORCH_XPU_API bool CachingHostAllocator_recordEvent(
+inline TORCH_XPU_API bool CachingHostAllocator_recordEvent(
```

Comment:
Why isn't this function taking a c10::Stream directly? I don't see it being used either within this repo?

Reply:
This is a device-specific API would be deprecated in the next [PR](https://github.com/pytorch/pytorch/pull/151437/files#diff-c4f88ef3cdd556a3cb9bbfd2524fbb7cb3815f5e13441f7de3eb5abcc6d6fcc0R17). We would like to recommend user to use a unified API `at::getHostAllocator(kXPU).record_event(...)` that accepts `c10::Stream` as input parameter.

Reply:
It is used in torch-xpu-ops repo for [CopyKernel](https://github.com/intel/torch-xpu-ops/blob/ac60f9b8fa8302b7073783de176e66800b3870c5/src/ATen/native/xpu/Copy.cpp#L252) to ensure non-blocking copy accuracy.